tools/metilene/metilene.xml

```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" encoding="utf-8"?>
+<tool id="metilene" name="metilene" version="0.2-4.4">
+    <description>
+        calling differentially methylated regions from
+        bisulfite sequencing data
+    </description>
+    
+    <requirements>
+        <requirement type="package" version="0.2-4" >metilene</requirement>
+        <requirement type="binary">Rscript</requirement>
+        <requirement type="package" version="5.18.1" >perl</requirement>
+        <requirement type="package" version="3.2.1" >R</requirement>
+        <requirement type="package" version="2.24" >bedtools</requirement>
+        <requirement type="package" version="8.22">gnu_coreutils</requirement>
+    </requirements>
+    
+    <stdio>
+        <exit_code range=":-1"/>
+        <exit_code range="1:" />
+    </stdio>
+
+    <version_command>echo "0.2-4"</version_command>
+
+    <command><![CDATA[
+    ## =============== preprocessing
+    ##   generate metilene_g1_g2.input from multiple bedgraph files
+    metilene_input.pl 
+    --in1 $opt_in1
```

Comment:
The test contains multiple inputs and is working. When I check this
(e.g. let this $opt_in be printed to stdout, it is comma-separated also
if I call from the web interface (using planemo serve). Are you sure,
this can be space-separated?

PS.: at this opportunity, I stumbled upon a bug in metilene itself. I
assume this is a bug in the output script; is there a (simple) way to
mark only the postprocessing results as failed in this case?


On 12/16/2015 08:55 AM, Björn Grüning wrote:
> In tools/metilene/metilene.xml
> <https://github.com/bgruening/galaxytools/pull/279#discussion_r47745409>:
> 
>> +        <requirement type="package" version="2.24" >bedtools</requirement>
>> +        <requirement type="package" version="8.22">gnu_coreutils</requirement>
>> +    </requirements>
>> +    
>> +    <stdio>
>> +        <exit_code range=":-1"/>
>> +        <exit_code range="1:" />
>> +    </stdio>
>> +
>> +    <version_command>echo "0.2-4"</version_command>
>> +
>> +    <command><![CDATA[
>> +    ## =============== preprocessing
>> +    ##   generate metilene_g1_g2.input from multiple bedgraph files
>> +    metilene_input.pl 
>> +    --in1 $opt_in1
> 
> |$opt_in1| is now space separated is this correct? Can this be handled
> if the user specifies n > 1 files?
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bgruening/galaxytools/pull/279/files#r47745409>.
> 


-- 
Dr. Sebastian Will
Bioinformatics/IZBI
Härtelstrasse 16-18, Room 448.1
University Leipzig
D-04107 Leipzig

Tel.: +49 (0) 341 97 16689